packages/layers/base/app/components/Connect/Tombstone/index.vue

```diff
@@ -0,0 +1,180 @@
+  }
+}
+
+const save = (sideDetail: ConnectTombstoneSideDetail) => {
+  if (sideDetail.edit) {
+    sideDetail.edit.validation?.validate(sideDetail.value)
+    if (sideDetail.edit.validation?.error) {
+      return
+    }
+    sideDetail.edit.isEditing = false
+    sideDetail.edit.action()
+  }
+}
+
+const handleButtonAction = async (button: ButtonProps, event: MouseEvent) => {
+  button.loading = true
+  if (button.onClick) {
+    if (Array.isArray(button.onClick)) {
+      await Promise.all(button.onClick.map(fn => fn(event)))
+    } else if (button.onClick) {
+      await button.onClick(event)
+    }
+  }
+  button.loading = false
+}
+</script>
+
+<template>
+  <div class="bg-white py-8 px-2 sm:px-4">
+    <div class="app-inner-container">
+      <div
+        v-if="tombstone.loading"
+        class="flex flex-col gap-2 *:space-y-2 sm:flex-row"
+      >
+        <div class="grow">
+          <USkeleton class="h-9 w-[400px] rounded" />
+          <USkeleton class="h-5 w-[250px] rounded" />
+          <USkeleton class="h-5 w-[200px] rounded" />
+          <USkeleton class="h-5 w-[150px] rounded" />
+        </div>
+        <div>
+          <USkeleton class="h-5 w-[300px] rounded" />
+          <USkeleton class="h-5 w-[300px] rounded" />
+          <USkeleton class="h-5 w-[300px] rounded" />
+          <USkeleton class="h-5 w-[300px] rounded" />
+        </div>
+      </div>
+      <div
+        v-else
+        class="flex flex-col gap-2 sm:flex-row"
+      >
+        <div class="grow space-y-4">
+          <div>
+            <component
+              :is="tombstone.title.el"
+              class="text-[1.375rem] font-bold text-neutral-highlighted"
+            >
+              {{ tombstone.title.text }}
+            </component>
+            <div
+              v-if="tombstone.subtitles.length"
+              class="flex divide-x divide-gray-500"
+            >
+              <ConnectTombstoneItem
+                v-for="subtitle, i in tombstone.subtitles"
+                :key="'subtitle-' + i"
+                v-bind="subtitle"
+                class="px-2"
+                :class="i === 0 ? 'pl-0' : ''"
+              />
+            </div>
+          </div>
+          <div class="space-y-1">
+            <slot name="details">
+              <div class="flex space-x-2">
+                <div
+                  v-for="detail, i in tombstone.details"
+                  :key="'detail-' + i"
+                >
+                  <ConnectTombstoneItem v-bind="detail" />
+                </div>
+              </div>
+            </slot>
+            <slot name="buttons">
+              <div
+                v-if="tombstone.bottomButtons.length"
+                class="flex flex-wrap gap-2"
+              >
+                <template v-for="btn in tombstone.bottomButtons" :key="btn.button?.label || btn.vNode?.key">
+                  <component :is="btn.vNode" v-if="btn.vNode" />
+                  <UButton
```

Comment:
The vNode is a cool thing I haven't used before. I think it would also be good to have a dynamic slot wrapping each button (i.e. <slot name=`button-${i}`>). That way I can pretty easily just add whatever I want in if one of the items is something special -- not sure how flexible or clean the vNode will be for everything

Reply:
Although your example is pretty awesome using vNode. We can always add a dynamic slot later if there's a need for it